### DISCUSSION TITLE
How to Replace the Value charsPerLine 25 in the New Method Printer.addTextLine?

### DISCUSSION THREAD

**Discussion by souhailtourjmen** [tr3v3r/react-native-esc-pos-printer#196] [Q&A] [answered]
Hello,

Thank you for the updates with the new Printer API. I have a question regarding the transition from the old textLine method to the new Printer.addTextLine method.

In the old method, I was using a numeric parameter like this:
`this.printing
  .textLine(25, {
    left: `\t+${supp.quantity > 0 ? supp?.quantity : supp.quantitytopay} ${supp.name}`,
    right: '--',
    gapSymbol: ' ',
  })
  .newline();
`
In the new API, I'm currently using:
`await Printer.addTextLine(this.printing, {
  left: `\t+${supp.quantity > 0 ? supp?.quantity : supp.quantitytopay} ${supp.name}`,
  right: '--',
  gapSymbol: ' ',
});`

 How can I integrate the value 25 in the new Printer.addTextLine method to achieve a similar result to the old method? This value was important for my formatting, and I can't find the equivalent in the new API.

Thank you in advance for your help!
![Uploading media_20241122_094604_2010511119239528091.jpg…]()

resolve methode  
file :react-native-esc-pos-printer/src/printer/printHelpers/addTextLine.ts

import { spaceBetween, getFontACharsPerLine } from '../utils';
import type { Printer } from '../Printer';
import type { SpaceBetweenParams } from '../types';
import { PrinterConstants, DEFAULT_PAPER_WIDTH } from '../constants';

export async function addTextLine(
  printer: Printer,
  params: SpaceBetweenParams,
  customCharsPerLine?: number

) {
  const printerCharsPerLinePerWidth = getFontACharsPerLine(printer.deviceName);
  const { value: paperWidth } = await printer.getPrinterSetting(
    PrinterConstants.PRINTER_SETTING_PAPERWIDTH
  );

  const charsPerLine = customCharsPerLine ? customCharsPerLine :
    printerCharsPerLinePerWidth[paperWidth || DEFAULT_PAPER_WIDTH];

  const text = spaceBetween(
    Math.ceil(charsPerLine / printer.currentFontWidth),
    params
  );
  await printer.addText(text);
}
file :react-native-esc-pos-printer/lib/typescript/src/printer/printHelpers/addTextLine.d.tsimport

type { Printer } from '../Printer';
import type { SpaceBetweenParams } from '../types';
export declare function addTextLine(printer: Printer, params: SpaceBetweenParams,customCharsPerLine?: number): Promise<void>;
//# sourceMappingURL=addTextLine.d.ts.map

**Comment by tr3v3r** [comment, marked as answer]
@souhailtourjmen I've removed this since this method was mostly for automatic layout detection.

You have two options: 

1) raise a PR with this option for [this function ](https://github.com/tr3v3r/react-native-esc-pos-printer/blob/main/src/printer/printHelpers/addTextLine.ts). Just add custom charsPerLine property. Then I can merge and release new version for you

2) you can switch to printing from React Native View method for better Developer experience

https://github.com/tr3v3r/react-native-esc-pos-printer/blob/main/docs/QUICK_START.md#2-print-from-react-native-view

**Reply by souhailtourjmen** [reply]
Thank you for the clarification! I understand why the method was removed, but I think raising a PR would be the best solution for my case. I have around 10 different ticket templates, so switching to React Native View isn't practical for me. 
I'll go ahead and prepare a PR with the new code for the custom charsPerLine. I'll send it over soon. Thanks for the support!
